go-controller/pkg/ovn/egressfirewall_test.go

```diff
@@ -692,9 +692,270 @@ var _ = ginkgo.Describe("OVN EgressFirewall Operations for local gateway mode",
+					&egressfirewallapi.EgressFirewallList{
+						Items: []egressfirewallapi.EgressFirewall{
+							*egressFirewall,
+						},
+					},
+					&v1.NamespaceList{
+						Items: []v1.Namespace{
+							namespace1,
+						},
+					},
+					&v1.NodeList{
+						Items: []v1.Node{
+							{
+								Status: v1.NodeStatus{
+									Phase: v1.NodeRunning,
+								},
+								ObjectMeta: newObjectMeta(node1Name, ""),
+							},
+						},
+					})
+
+				fakeOVN.controller.WatchNamespaces()
+				fakeOVN.controller.WatchEgressFirewall()
+
+				ipv4ACL := libovsdbops.BuildACL(
+					"",
+					t.DirectionToLPort,
+					t.EgressFirewallStartPriority,
+					"(ip4.dst == 1.2.3.4/23) && ip4.src == $a10481622940199974102 && ip4.dst != 10.128.0.0/14",
+					nbdb.ACLActionAllow,
+					"",
+					"",
+					false,
+					map[string]string{"egressFirewall": "namespace1"},
+					nil,
+				)
+				ipv4ACL.UUID = "ipv4ACL-UUID"
+
+				// new ACL will be added to the switch
+				finalNodeSwitch := &nbdb.LogicalSwitch{
+					UUID: InitialNodeSwitch.UUID,
+					Name: InitialNodeSwitch.Name,
+					ACLs: []string{ipv4ACL.UUID},
+				}
+
+				// new ACL will be added to the switch
+				expectedDatabaseState := []libovsdb.TestData{
+					ipv4ACL,
+					finalNodeSwitch,
+					clusterRouter,
+				}
+
+				gomega.Expect(fakeOVN.nbClient).To(libovsdbtest.HaveData(expectedDatabaseState))
+
+				_, err := fakeOVN.fakeClient.EgressFirewallClient.K8sV1().EgressFirewalls(egressFirewall.Namespace).Get(context.TODO(), egressFirewall.Name, metav1.GetOptions{})
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				ginkgo.By("Bringing down NBDB")
+				// inject transient problem, nbdb is down
+				fakeOVN.controller.nbClient.Close()
+				gomega.Eventually(func() bool {
+					return fakeOVN.controller.nbClient.Connected()
+				}).Should(gomega.BeFalse())
+
+				_, err = fakeOVN.fakeClient.EgressFirewallClient.K8sV1().EgressFirewalls(egressFirewall1.Namespace).Update(context.TODO(), egressFirewall1, metav1.UpdateOptions{})
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				// sleep long enough for TransactWithRetry to fail, causing egress firewall Add to fail
+				time.Sleep(t.OVSDBTimeout + time.Second)
+				// check to see if the retry cache has an entry for this egress firewall
+				key := getEgressFirewallNamespacedName(egressFirewall)
+				gomega.Eventually(func() *retryObjEntry {
+					return fakeOVN.controller.retryEgressFirewalls.getObjRetryEntry(key)
+				}).ShouldNot(gomega.BeNil())
+				connCtx, cancel := context.WithTimeout(context.Background(), t.OVSDBTimeout)
+				defer cancel()
+				resetNBClient(connCtx, fakeOVN.controller.nbClient)
+				fakeOVN.controller.retryEgressFirewalls.requestRetryObjs()
+				ipv4ACL.Action = nbdb.ACLActionDrop
+
+				// check the cache no longer has the entry
+				gomega.Eventually(func() *retryObjEntry {
+					return fakeOVN.controller.retryEgressFirewalls.getObjRetryEntry(key)
+				}).Should(gomega.BeNil())
+
```

Comment:
missing a check for the expected data post update:

gomega.Eventually(fakeOVN.nbClient).Should(libovsdbtest.HaveData(expectedDatabaseState))